calico_containers/tests/st/utils.py

```diff
@@ -17,14 +17,22 @@ def cleanup_inside(name):
     """
     Clean the inside of a container by deleting the containers and images within it.
     """
-    docker("exec", "-t", name, "bash", "-c",
-           "docker rm -f $(docker ps -qa) ; docker rmi $(docker images -qa)",
-           _ok_code=[0,
-                     1,  # Caused by 'docker: "rm" requires a minimum of 1 argument.' et al.
-                     127,  # Caused by '"docker": no command found'
-                     255,  # Caused by '"bash": executable file not found in $PATH'
-                    ]
-          )
+    try:
+        inner_containers = docker("exec", "-t", name, "bash", "-c", "docker ps -qa").stdout.split()
+        for container in inner_containers:
+            docker("exec", "-t", name, "bash", "-c", "docker rm -f %s" % container)
+
+        inner_images = docker("exec", "-t", name, "bash", "-c", "docker images -q").stdout.split()
+        for image in inner_images:
+            docker("exec", "-t", name, "bash", "-c", "docker rmi %s" % image)
+    except ErrorReturnCode_127:
```

Comment:
@fasaxc Yeah, I wondered whether I should do that. Both caught exceptions are from cases that happen on the first line (one comment is wrong). If an `rm` or `rmi` fails, the exception won't be caught, which is what I'm looking for; then I can go investigate why that happened. It *could* be the case that a `127` or `255` might be raised by some other means, which will then be caught and fail to cleanup the rest of the loop. I feel this is unlikely given that all the commands are so similar. Restructuring it to guarantee catching all also makes it a fair bit messier. Do you think I'm taking the wrong side of the trade-off?